File: Front-end/src/app/models/account.ts
import { Category } from "./category";
import { Expense } from "./expense";

export class Account {
    account_id: number;
    account_created: string;
    account_name: string;
    account_deleted: string;

    expenseList: Expense[];
    categoryList: Category[];

    constructor(account_id: number,
        account_created: string,
        account_name: string,
        account_deleted: string,
        expenseList: Expense[],
        categoryList: Category[]) {
            this.account_created = account_created;
            this.account_id = account_id;
            this.account_name = account_name;
            this.account_deleted = account_deleted;
            this.expenseList = expenseList;
            this.categoryList =  categoryList; 
        }
}

File: Back-end/src/main/java/com/project/budgetapp/models/Account.java
package com.project.budgetapp.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.List;

@Entity(name = "accounts")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long account_id;
    private String account_created;
    @NotEmpty
    private String account_name;
    private String account_deleted;

    @OneToMany(mappedBy = "account_id", cascade = CascadeType.REMOVE)
    private List<Category> categoryList;

    public Account() {}

    public List<Category> getCategoryList() {
        return categoryList;
    }

    public void setCategoryList(List<Category> categoryList) {
        this.categoryList = categoryList;
    }

    public Long getAccount_id() {
        return account_id;
    }

    public void setAccount_id(Long account_id) {
        this.account_id = account_id;
    }

    public String getAccount_created() {
        return account_created;
    }

    public void setAccount_created(String account_created) {
        this.account_created = account_created;
    }

    public String getAccount_name() {
        return account_name;
    }

    public void setAccount_name(String account_name) {
        this.account_name = account_name;
    }

    public String getAccount_deleted() {
        return account_deleted;
    }

    public void setAccount_deleted(String deleted) {
        this.account_deleted = deleted;
    }
}
